TaskServiceInterview.cs

```diff
@@ -0,0 +1,448 @@
+        private readonly IInternalReminderFinder internalReminderFinder;
+        private readonly ITaskCreateService taskCreateService;
+        private readonly IRepository<TaskConnection> taskConnectionRepository;
+
+        private string taskPostponed = "Task postponed";
+        private string taskFinished = "Task finished";
+        private string sentence = "Ownership taken";
+
+        public TaskServiceInterview(
+            IMappingServant mappingServant,
+            IRepository<TaskEntity> repository,
+            ITaskFinder taskFinder,
+            IRepository<Classification> classificationRepository,
+            IAuditLogServant auditLogServant,
+            ITaskLockServant taskLockServant,
+            ITaskChangeNotificationServant taskChangeNotificationServant,
+            IClock clock,
+            IPermissionServant permissionServant,
+            IClassificationAccessServant classificationAccessServant,
+            IValidator<TaskEntity> taskValidator,
+            IAsynchronousOperator asynchronousOperator,
+            ICustomerDetailsServant customerDetailsServant,
+            IBusinessCalendarClient businessCalendarClient,
+            IPsbConnector psbConnector,
+            INotificationService notificationService,
+            ITaskBusinessEventFinder taskBusinessEventFinder,
+            IInternalReminderFinder internalReminderFinder,
+            ITaskCreateService taskCreateService,
+            IRepository<TaskConnection> taskConnectionRepository)
+        {
+            this.notificationService = notificationService;
+            this.taskBusinessEventFinder = taskBusinessEventFinder;
+            this.internalReminderFinder = internalReminderFinder;
+            this.taskCreateService = taskCreateService;
+            this.taskFinder = taskFinder;
+            this.classificationRepository = classificationRepository;
+            this.mappingServant = mappingServant;
+            this.repository = repository;
+            this.auditLogServant = auditLogServant;
+            this.taskLockServant = taskLockServant;
+            this.taskChangeNotificationServant = taskChangeNotificationServant;
+            this.clock = clock;
+            this.classificationRepository = classificationRepository;
+            this.permissionServant = permissionServant;
+            this.classificationAccessServant = classificationAccessServant;
+            this.taskValidator = taskValidator;
+            this.asynchronousOperator = asynchronousOperator;
+            this.businessCalendarClient = businessCalendarClient;
+            this.customerDetailsServant = customerDetailsServant;
+            this.psbConnector = psbConnector;
+            this.taskConnectionRepository = taskConnectionRepository;
+        }
+
+        public TaskDetailsItem GetById(Guid id, string employeeInitials)
+        {
+            var task = this.repository.Single(id, t => t.FlexType,

```

Comment:
Try to use SingleOrDefault to get null object instead of an exception. Remember to handle output before if statement (task.).